### DISCUSSION TITLE
VAE accuracy/confidence?

### DISCUSSION THREAD

**Discussion by jelrod14** [DrCoffey/DeepSqueak#221] [General]
Hello, 
I was just wondering if there was a way to pull a performance metric from the contour invariant VAE model after clustering that says how confident the model is in putting a particular USV into a certain cluster. For example, if my clustering yields 8 clusters, is there a range as to how "cluster 1" a call is and towards the end of that range things get more variable? If it's possible to know and pull this metric from deepsqueak that would be super helpful!
Thanks!

**Comment by DrCoffey** [comment]
Hey @jelrod14, that information is definitely available but not easily accessible. Line 72-77 of UnsupervisedClustering_Callback.m is:

[clustAssign,D] = knnsearch(C,data,'Distance','euclidean');
            
            %% Sort the calls by how close they are to the cluster center
            [~,idx] = sort(D);
            clustAssign = clustAssign(idx);
            ClusteringData = ClusteringData(idx,:);

D is the distance to the centroid of the assigned cluster. You could try to add D to the ClusteringData table and then it would get saved along with the clustering. 

I won't have time to add this and test it etc for a little while, but if you change the code first you can always submit a pull request to get it added to the main code.

-Kevin
